static/js/permits_complementary_documents.js

```diff
@@ -0,0 +1,80 @@
+let container = document.getElementById('complementary-documents')
+let documents = document.getElementsByClassName('complementary-document')
+let nbDocuments = documents.length;
+
+hideChildrenDocumentTypes(document)
+
+document.getElementById('add-document').addEventListener('click', (e) => {
+  e.preventDefault()
+
+  // display the remove button
+  document.getElementById('remove-document').removeAttribute('hidden')
+  // increase the number of documents
+  ++nbDocuments
+
+  // regex to find all instances of the form number
+  let regex = RegExp(`form-(\\d){1}-`,'g')
+
+  // clone the form
+  let clone = documents[0].cloneNode(true)
+  clone.id = `document-${nbDocuments-1}`
+  clone.innerHTML = clone.innerHTML.replace(regex, `form-${nbDocuments-1}-`)
+
+  // change the "form number"
+  document.getElementById("id_form-TOTAL_FORMS").value = nbDocuments
+
+  hideChildrenDocumentTypes(clone)
+
+  clone.prepend(document.createElement('hr'))
+  container.appendChild(clone)
+})
+
+document.getElementById('remove-document').addEventListener('click', (e) => {
+  e.preventDefault()
+
+  // we don't want to delete the last form
+  if (nbDocuments === 1) {
+    return
+  }
+
+  // decrease the number of documents
+  --nbDocuments
+
+  // we don't need the remove button anymore
+  if (nbDocuments === 1) {
+    e.target.setAttribute('hidden', '')
+  }
+
+  // update the number of total forms
+  document.getElementById("id_form-TOTAL_FORMS").value = nbDocuments
+
+  // remove the last form
+  let target = [...documents].pop()
+  target.remove()
+})
+
+document.addEventListener('change', (e) => {
```

Comment:
Could you not add the event listener directly on the document_type select instead of checking that we're working with the correcte element in the next few lines of code?

Reply:
The problem comes from the extra forms that can be created afterwards. If I use `.addEventListener('click'..)` directly on the select, it won't apply to the selects that get created afterwards